src/spark_rapids_ml/core.py

```diff
@@ -531,7 +542,7 @@ def _pre_process_data(
         return select_cols, multi_col_names, dimension, feature_type
 
 
-class _CumlModel(Model, _CumlCommon, _CumlParams):
+class _CumlModel(Model, _CumlParams, _CumlCommon):
```

Comment:
Why order change here?

Reply:
Long story short, this relates to python MRO (Method Resolution Order). 

Specifically, order change here to keep the same inheritance order as in _CumlCaller(_CumlParams, _CumlCommon). Otherwise, a python MRO  error will appear. 

We must use inheritance order _CumlCaller(_CumlParams, _CumlCommon) because reverse order _CumlCaller(_CumlCommon, _CumlParams) triggers a MRO error in _CumlEstimator(Estimator, _CumlCaller). The error message is "TypeError: Cannot create a consistent method resolution order (MRO) for bases Params, Generic".  It seems because in _CumlCaller superclass Generic appears before superclass Params. But in Estimator superclass Params appears before the superclass Generic.